events/emitter.go

```diff
@@ -0,0 +1,62 @@
+package events
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containerd/containerd/api/types/event"
+	"github.com/containerd/containerd/namespaces"
+	goevents "github.com/docker/go-events"
+)
+
+const (
+	EventVersion = "v1"
+)
+
+type Emitter struct {
+	sinks       map[string]*eventSink
+	broadcaster *goevents.Broadcaster
+	m           *sync.Mutex
```

Comment:
I'm unclear on what the mutex is used for. It doesn't seem to be protecting `sinks`, because the presence in the map is always tested before acquiring the mutex. But I don't think any locking is needed for `broadcaster.Add` and `broadcaster.Remove`.